AK/FixedArray.h

```diff
@@ -36,6 +36,47 @@ class FixedArray {
         return MUST(try_create(size));
     }
 
+    // NOTE:
+    // Below here, we do a little dance where we have two publicly accessible templated functions
+    // which are simply always-inlined wrappers around private helper functions.
+    // The reason is that the publicly accessable templated functions are templated on the
+    // size of the C-style array from which they allow construction. The private helper functions
+    // instead take the size as a runtime argument. We do not want to expose these helper functions
+    // publicly because we do not want to allow for a user to pass the wrong size to us. The reason
+    // that we always-inline the wrappers is that it does not make sense to have a version of them
+    // for each possible size since the code will be essentially identical in every case.
+
+private:
+    static ErrorOr<FixedArray<T>> try_create(T(&&array)[], size_t size)
+    {
+        if (size == 0)
+            return FixedArray<T>();
+        T* elements = static_cast<T*>(kmalloc_array(size, sizeof(T)));
+        if (!elements)
+            return Error::from_errno(ENOMEM);
+        for (size_t i = 0; i < size; ++i)
+            new (&elements[i]) T(move(array[i]));
+        return FixedArray<T>(size, elements);
+    }
```

Comment:
This can just live in `ErrorOr<FixedArray<T>> try_create(T(&&array)[size])`, why have an extra function lying around when all its uses are going to be inlined? (this looks small enough to be inlined itself tbh, and at least GCC agrees: https://serenity.godbolt.org/z/x1E6oGh89)

Reply:
I think the paragraph above explains that clearly. If we do not extract this out into a helper function, there will be a version of this function for every value of the `size` template argument, which is completely redundant, because the code will be identical in all cases except for the constant.

Reply:
- that function is small enough to be inlined
- I just gave you an example of it being inlined and not even emitted

Reply:
Isn't it still good to be explicit about this?

Reply:
considering there's an 8-line explanation attached for no actual change, I don't know if it is.

Reply:
Okay, I will remove the wrappers. I'll at least leave a note about this so that a future reader will know that he doesn't have to worry about this.